src/compat/posix/time/strptime.c

```diff
@@ -0,0 +1,334 @@
+/**
+ * @file strptime.c
+ * @brief writing linux function strptime for embox os
+ *
+ * author Tamara Deryugina <tamara.deryugina@gmail.com>
+ * @date 02.03.2020
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <ctype.h>
+#include <math.h>
+
+static int num_length(const char *s) {
+	int count = 0;
+	while ((s[count] >= '0') && (s[count] <= '9')) {
+		count++;
+	}
+	return count;
+}
+
+static int word_length(const char *s) {
+	int count = 0;
+	while ((s[count] >= 'A') && (s[count] <= 'z')) {
+		count++;
+	}
+	return count;
+}
+
+static int cmp_before_percent(const char *s, const char *format) {
+	int count = 0;
+	while (format[count] != '\0') {
+		if (format[count] == '%') {
+			return count;
+		}
+		if (strncmp(&format[count], &s[count], 1) != 0) {
+			return 0;
+		}
+		count++;
+	}
+	return count;
+}
+
+static int cmp_symb(const char *s, const char *format, struct tm *tm, int w_leng) {
+	int count = 2;
+	int month = 0;
+	int wday = 0;
+	char arr_day[7][10] = {"sunday", "monday", "tuesday", "wednesday", "thursday", "friday", "saturday"};
+	char arr_month[13][10] = {"january", "february", "march", "april", "may", "june",
+		"july", "august", "september", "october", "november", "december"};
+
+	if ((format[0] == 'a') || (format[0] == 'A')) {
+		while (wday != 7) {
+			if ((tolower(s[0]) == arr_day[wday][0]) && (tolower(s[1]) == arr_day[wday][1])) {
```

Comment:
Why do you compare only two first characters here?

Reply:
I had idea: firstly to find the day of the week with help of first two characters and after compare all string symbols with symbols of defined day of the week. But now I understand that use nested loop is more compact way than my. So I rewrite this part of code.
Thank you!